modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java

```diff
@@ -267,14 +266,23 @@ public void gatherLocalObjectStatisticsAsync(
     /** {@inheritDoc} */
     @Override public StatisticsGatheringFuture<Map<StatisticsTarget, ObjectStatistics>>[] gatherObjectStatisticsAsync(
         StatisticsTarget... keys
-    ) throws IgniteCheckedException {
-        checkSupport("collect statistics async");
+    ) {
+
+        Set<StatisticsKeyMessage> keysMsg = Arrays.stream(keys).map(StatisticsUtils::statisticsKeyMessage)
+            .collect(Collectors.toSet());
 
-        Set<StatisticsKeyMessage> keysMsg = Arrays.stream(keys).map(
-            t -> new StatisticsKeyMessage(t.schema(), t.obj(), Arrays.asList(t.columns()))).collect(Collectors.toSet());
         Map<CacheGroupContext, Collection<StatisticsKeyMessage>> grpsKeys = helper.splitByGroups(keysMsg);
 
-        List<StatisticsGatheringFuture<Void>> res = new ArrayList<>();
+        if (!IgniteFeatures.allNodesSupport(ctx, IgniteFeatures.STATISTICS_COLLECTION, IgniteDiscoverySpi.SRV_NODES)) {
```

Comment:
it's better to check this with `checkStatisticsSupport()`, but slightly modify it to take param to decide wether an exception should be thrown or not (`true` by default)

```
 void checkStatisticsSupport(String op) {
     checkStatisticsSupport(op, true)
 }

boolean checkStatisticsSupport(String op, boolean shouldThrow) {
     if (<support>)
         return true;
     
     if (shouldThrow)
         throw new Ex();

     return false;
 } 
```

Reply:
done